tools/clang/lib/SPIRV/InitListHandler.cpp

```diff
@@ -399,39 +400,52 @@ InitListHandler::createInitForStructType(QualType type, SourceLocation srcLoc,
     tryToSplitStruct();
   }
 
-  llvm::SmallVector<SpirvInstruction *, 4> fields;
-
-  // Initialize base classes first.
-  llvm::SmallVector<SpirvInstruction *, 4> base_fields;
-  const RecordDecl *structDecl = type->getAsStructureType()->getDecl();
-  if (auto *cxxStructDecl = dyn_cast<CXXRecordDecl>(structDecl)) {
-    for (CXXBaseSpecifier base : cxxStructDecl->bases()) {
-      QualType baseType = base.getType();
-      const RecordType *baseStructType = baseType->getAsStructureType();
-      if (baseStructType == nullptr) {
-        continue;
-      }
-      const RecordDecl *baseStructDecl = baseStructType->getDecl();
-      for (const auto *field : baseStructDecl->fields()) {
-        base_fields.push_back(
-            createInitForType(field->getType(), field->getLocation(), range));
-        if (!base_fields.back())
-          return nullptr;
-      }
-      fields.push_back(spvBuilder.createCompositeConstruct(
-          baseType, base_fields, srcLoc, range));
-      base_fields.clear();
-    }
-  }
+  const RecordType *recordType = type->getAs<RecordType>();
+  assert(recordType);
 
-  for (const auto *field : structDecl->fields()) {
-    fields.push_back(
-        createInitForType(field->getType(), field->getLocation(), range));
-    if (!fields.back())
-      return nullptr;
-  }
+  LowerTypeVisitor lowerTypeVisitor(astContext, theEmitter.getSpirvContext(),
+                                    theEmitter.getSpirvOptions());
+  const SpirvType *spirvType =
+      lowerTypeVisitor.lowerType(type, SpirvLayoutRule::Void, false, srcLoc);
 
-  // TODO: use OpConstantComposite when all components are constants
+  llvm::SmallVector<SpirvInstruction *, 4> fields;
+  forEachSpirvField(
+      recordType, dyn_cast<StructType>(spirvType),
```

Comment:
Should you assert that this dyn_cast is not nullptr?

Reply:
done